### PR TITLE
[BugFix] No check on success of protobuf serialization

### PR DESCRIPTION
**Pull request by sduzh**
Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 32 / 36 (88.89%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/lake/tablet_manager.cpp|16|19|84.21%|[519, 953, 954]|
|:large_blue_circle:|src/storage/protobuf_file.cpp|10|11|90.91%|[68]|
|:large_blue_circle:|src/storage/lake/meta_file.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/protobuf_file.h|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/delta_column_group.cpp|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/tablet_meta.cpp|2|2|100.00%|[]|

File: be/src/storage/lake/tablet_manager.cpp [hunk 1]
```diff
@@ -424,11 +417,11 @@ Status TabletManager::put_tablet_metadata(const TabletMetadata& metadata) {
...
     auto t0 = butil::gettimeofday_us();
-    MetaFileReader reader(metadata_location, fill_cache);
-    RETURN_IF_ERROR(reader.load());
-    auto res = reader.get_meta();
+    auto metadata = std::make_shared<TabletMetadataPB>();
+    ProtobufFile file(metadata_location);
+    RETURN_IF_ERROR(file.load(metadata.get(), fill_cache));
     g_get_tablet_metadata_latency << (butil::gettimeofday_us() - t0);
-    return res;
+    return std::move(metadata);
```
Comment (wyb) [inline be/src/storage/lake/tablet_manager.cpp:424]:
```suggestion
    return metadata;
```

File: be/src/storage/protobuf_file.cpp [hunk 2]
```diff
@@ -93,4 +101,39 @@ Status ProtobufFile::load(::google::protobuf::Message* message) {
     return Status::OK();
 }
 
+Status ProtobufFile::save(const ::google::protobuf::Message& message, bool sync) {
+    std::string serialized_message;
+    bool r = message.SerializeToString(&serialized_message);
+    TEST_SYNC_POINT_CALLBACK("ProtobufFile::save:serialize", &r);
+    if (UNLIKELY(!r)) {
+        return Status::InternalError("failed to serialize protobuf to string, maybe the protobuf is too large");
```
Comment (wyb) [inline be/src/storage/protobuf_file.cpp:109]: better to add path in the message.

File: be/src/storage/lake/tablet_manager.cpp (hunk 1 above)
Comment (sduzh) [reply be/src/storage/lake/tablet_manager.cpp:424]: any reason?

File: be/src/storage/protobuf_file.cpp (hunk 2 above)
Comment (sduzh) [reply be/src/storage/protobuf_file.cpp]: updated

File: be/src/storage/lake/tablet_manager.cpp (hunk 1 above)
Comment (wyb) [reply be/src/storage/lake/tablet_manager.cpp:424]: auto return value optimization？

Comment (sduzh) [reply be/src/storage/lake/tablet_manager.cpp:424]: > auto return value optimization？

The function return value is of type StatusOr<std::shared_ptr<TabletMetadataPB>>, not std::shared_ptr<TabletMetadataPB>

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 72 / 79 (91.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/protobuf_file.cpp|44|51|86.27%|[76, 85, 93, 94, 97, 100, 128]|
|:large_blue_circle:|src/storage/lake/meta_file.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/tablet_manager.cpp|21|21|100.00%|[]|
|:large_blue_circle:|src/storage/protobuf_file.h|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/delta_column_group.cpp|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/tablet_meta.cpp|2|2|100.00%|[]|

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 72 / 79 (91.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/protobuf_file.cpp|44|51|86.27%|[76, 85, 93, 94, 97, 100, 128]|
|:large_blue_circle:|src/storage/lake/meta_file.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/tablet_manager.cpp|21|21|100.00%|[]|
|:large_blue_circle:|src/storage/protobuf_file.h|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/delta_column_group.cpp|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/tablet_meta.cpp|2|2|100.00%|[]|

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**